Makefile

```diff
@@ -0,0 +1,29 @@
+test:
+	go test -v ./...
+
+build: test
+	go build
+
+check: lint vet fmtcheck ineffassign
+
+lint:
+	golint -set_exit_status .
+
+vet:
+	go vet
+
+fmtcheck:
```

Comment:
This would be better as a precommit hook. Here's one you can use as a model:

https://golang.org/misc/git/pre-commit

Reply:
Having a pre-commit hook will not guarantee that the code is well formatted: Some tools by-pass the hooks, committing from GoLand for example.

Having such a check performed in Travis for every pull request will, imho, be a surefire guarantee that the proposed changes are well formatted.